pkg/hintrunner/zero/zerohint_dictionaries.go

```diff
@@ -154,6 +154,81 @@ func createSquashDictInnerAssertLenKeysHinter() (hinter.Hinter, error) {
 	return newSquashDictInnerAssertLenKeysHint(), nil
 }
 
+// SquashDictInnerCheckAccessIndex hint updates the access index
+// during the dictionary squashing process
+//
+// `newSquashDictInnerCheckAccessIndexHint` takes 1 operander as argument
+//   - `loopTemps` variable is a struct containing `index_delta_minus1` as
+//     a first field
+//
+// `newSquashDictInnerCheckAccessIndexHint` writes to the first field `index_delta_minus1`
+// of the `loop_temps` struct the result of `new_access_index - current_access_index - 1`
+// and assigns `new_access_index` to `current_access_index` in the scope
+func newSquashDictInnerCheckAccessIndexHint(loopTemps hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "SquashDictInnerCheckAccessIndex",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> new_access_index = current_access_indices.pop()
+			//> ids.loop_temps.index_delta_minus1 = new_access_index - current_access_index - 1
+			//> current_access_index = new_access_index
+
+			currentAccessIndices_, err := ctx.ScopeManager.GetVariableValue("current_access_indices")
+			if err != nil {
+				return err
+			}
+
+			currentAccessIndices, ok := currentAccessIndices_.([]fp.Element)
+			if !ok {
+				return fmt.Errorf("casting currentAccessIndices_ into an array of felts failed")
+			}
+
+			newAccessIndex, err := utils.Pop(&currentAccessIndices)
+			if err != nil {
+				return err
+			}
+
+			currentAccessIndex_, err := ctx.ScopeManager.GetVariableValue("current_access_index")
+			if err != nil {
+				return err
+			}
+
+			currentAccessIndex, ok := currentAccessIndex_.(fp.Element)
+			if !ok {
+				return fmt.Errorf("casting currentAccessIndices_ into an array of felts failed")
```

Comment:
you mean into a felt not an array of felts

Reply:
true thanks !